base/tuple.jl

```diff
@@ -73,14 +73,15 @@ keys(@nospecialize t::Tuple) = OneTo(length(t))
 prevind(@nospecialize(t::Tuple), i::Integer) = Int(i)-1
 nextind(@nospecialize(t::Tuple), i::Integer) = Int(i)+1
 
-function keys(t::Tuple, t2::Tuple...)
-    @inline
-    OneTo(_maxlength(t, t2...))
+@noinline function throw_eachindex_mismatch_indices(::Tuple{}, inds...)
+    throw(DimensionMismatch("all inputs to eachindex must have the same length, got $(join(inds, ", ", " and "))"))
 end
-_maxlength(t::Tuple) = length(t)
-function _maxlength(t::Tuple, t2::Tuple, t3::Tuple...)
+
+function keys(t::Tuple, t2::Tuple...)
     @inline
-    max(length(t), _maxlength(t2, t3...))
+    lent = length(t)
+    all(==(lent), t2) || throw_eachindex_mismatch_indices((), t, t2...)
```

Comment:
This does not look correct: It has to compare to the lengths of the tuples in `t2`.
Maybe also add a test that verifies `eachindex((1,), (1,))` does `not` throw dimension mismatch?

Reply:
what do you mean?